xdsl/transforms/experimental/hls_convert_stencil_to_ll_mlir.py

```diff
@@ -0,0 +1,150 @@
+from dataclasses import dataclass
+
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    GreedyRewritePatternApplier,
+    op_type_rewrite_pattern,
+)
+from xdsl.ir import MLContext, Operation, OpResult
+from xdsl.dialects.builtin import i32, f64
+from xdsl.dialects.func import FuncOp, Call
+from xdsl.dialects import arith, builtin
+from xdsl.dialects.arith import Constant
+
+from xdsl.dialects.stencil import ExternalLoadOp
+
+from xdsl.dialects.experimental.hls import (
+    HLSStream,
+)
+
+from xdsl.dialects.memref import MemRefType
+
+from xdsl.passes import ModulePass
+
+from xdsl.utils.hints import isa
+
+from xdsl.dialects.llvm import LLVMPointerType, LLVMStructType, LLVMArrayType
+
+from xdsl.ir.core import BlockArgument
+
+
+# TODO docstrings and comments
```

Comment:
please address TODO

Reply:
I think this is fine for an experimental dialect, no?